plugins/GSdx/GS.cpp

```diff
@@ -457,39 +457,34 @@ EXPORT_C_(int) GSopen2(void** dsp, uint32 flags)
 	static bool stored_toggle_state = false;
 	bool toggle_state = !!(flags & 4);
 
-	int renderer = s_renderer;
+	GSRendererType renderer = s_renderer;
 	// Fresh start up or config file changed
-	if (renderer == -1)
+	if (renderer == GSRendererType::Undefined)
 	{
-#ifdef __linux__
-		// Use ogl renderer as default otherwise it crash at startup
-		// GSRenderOGL only GSDeviceOGL (not GSDeviceNULL)
-		renderer = theApp.GetConfig("Renderer", 12);
-#else
-		renderer = theApp.GetConfig("Renderer", 0);
-#endif
+		renderer = static_cast<GSRendererType>(theApp.GetConfig("Renderer", static_cast<int>(GSRendererType::Default)));
 	}
 	else if (stored_toggle_state != toggle_state)
 	{
 #ifdef _WIN32
-		int best_sw_renderer = GSUtil::CheckDirect3D11Level() >= D3D_FEATURE_LEVEL_10_0 ? 4 : 1; // dx11 / dx9 sw
+		GSRendererType best_sw_renderer = GSUtil::CheckDirect3D11Level() >= D3D_FEATURE_LEVEL_10_0 ? GSRendererType::DX1011_SW : GSRendererType::DX9_SW; // dx11 / dx9 sw
 
-		switch(renderer){
+		switch (renderer) {
 			// Use alternative renderer (SW if currently using HW renderer, and vice versa, keeping the same DX level)
-			case 1: renderer = 0; break; // DX9:  SW to HW
-			case 0: renderer = 1; break; // DX9:  HW to SW
-			case 4: renderer = 3; break; // DX11: SW to HW
-			case 3: renderer = 4; break; // DX11: HW to SW
-			case 13: renderer = 12; break; // OGL: SW to HW
-			case 12: renderer = 13; break; // OGL: HW to SW
-			default: renderer = best_sw_renderer; // If wasn't using DX (e.g. SDL), use best SW renderer.
+		case GSRendererType::DX9_SW: renderer = GSRendererType::DX9_HW; break; // DX9:  SW to HW
+		case GSRendererType::DX9_HW: renderer = GSRendererType::DX9_SW; break; // DX9:  HW to SW
+		case GSRendererType::DX1011_SW: renderer = GSRendererType::DX1011_HW; break; // DX11: SW to HW
```

Comment:
I think the enum in the present form better represents the actual code. Even if readability is reduced - from a coding point of view this naming convention seems to me more appropriate. I am not strongly forcing the current names but it seems you are also not strongly forcing your opinion. So it is even. I waited for other responses. turtleli said he doesn't care and current solutions seems ok. So I would leave it like that.

To be honest I wouldn't know if I should choose DX10 or DX11. DX10 seems to be the actual api that is used. DX11 is only used in something like backwards compatibility mode. Still most of the current users would see Direct3D11 in their gpu configuration. So DX10 seems to provide necessary functions and DX11 is more often seen by users. Deciding which one to chose is hard. So I just took the compromise.

And to be honest. Nobody really cares for this enum.